### ISSUE TITLE
Force HTTPS on final redirect from oauth2 middleware

### ISSUE THREAD

**Issue by TenSt** [dapr/components-contrib#486]
**Case:** A web-app which is exposed over HTTPS and access API that are behind darp + oauth2 middleware.

**Problem:** 'Location' header in the final redirect from dapr oauth2 middleware contains HTTP and redirect request is blocked by browser with mixed-content error.

**Solution:** add new key to metadata. If key is true - force HTTPS on final redirect to not get blocked by browser.

**Justification:** A case when web-app communicates with api or some other backend over HTTPS is pretty common. dapr does a very good job on protecting API, especially in conjunction with Oauth2 middleware. The issue is that dapr doesn't operate over HTTPS. HTTPS request to dapr can be handled by Nginx tls configuration or something similar (it verifies that connection is secure and then changes scheme to HTTP), but in the final redirect dapr oauth2 middleware returns 'Location' header with 'http' as a scheme and that leads to mixed-content errors in browser.

**Code**: I already have a POC written and verified in the k8s environment, so can create pull request to cover this.

**Comment by yaron2** [comment]
Thanks for raising this, a PR would be welcome.

**Comment by TenSt** [comment]
Thanks @yaron2. I've created new PR and issue in the docs repo to cover it.
